lib/ex_unit/lib/ex_unit/assertions.ex

```diff
@@ -7,6 +7,7 @@ defmodule ExUnit.AssertionError do
 
   defexception left: @no_value,
                right: @no_value,
+               mailbox: @no_value,
```

Comment:
Instead of adding a new key, maybe we can do context: {:mailbox, pins, mailbox}?

Reply:
Another option is to expose the relevant APIs in ExUnit.Diff so we can compute the diff from `assertions` instead of having to pass all information inside.

Reply:
The problem with exposing `ExUnit.Diff` is that `ExUnit.Formatter` has knowledge if the diffs actually should run or not, the colors that they should be in, and they also have some timeouts for the diff. I don't if it's worth to duplicate all those things or just ignore them.

I can change the context to something like `{:mailbox, pins, pattern, mailbox}` and change the `ExUnit.Diff` to return a list of diffs when doing that, but I would need to change the timeout in the formatter to be multiplied by the number of messages. Does this makes sense?

Reply:
Good points!

> I can change the context to something like {:mailbox, pins, pattern, mailbox} and change the ExUnit.Diff to return a list of diffs when doing that

You can pre-process the assertionerror struct and convert it into the structure you have today where you pass mailbox and `{:match, pins}`. My only goal is to avoid adding more fields to the future. But multiplying the diff per number of messages is also fine. :+1: